# *Análisis y limpieza de datos*

### Librerias necesarias

In [220]:
import pandas as pd
from Estandarizar import estandarizacion
from Estandarizar_telefono import limpiar_numero

### Lectura del archivo como un dataframe

In [221]:
df = pd.read_excel('Datos de alumnos y padres.xlsx')

### 

### Dimension

In [222]:
print("Cantidad de filas: " + str(df.shape[0]) + " | Cantidad de columnas: " + str(df.shape[1]))

Cantidad de filas: 431 | Cantidad de columnas: 52


### Columnas

In [223]:
print(df.columns)

Index(['clave', 'digito', 'digban', 'nombrea', 'sexo', 'domal', 'coloa',
       'pobla', 'cpa', 'fecnaca', 'naciona', 'tela', 'nivel', 'grado', 'grupo',
       'escpro', 'porcol', 'porins', 'codigo', 'madre', 'ocupam', 'emprem',
       'celularm', 'telofm', 'emailm', 'padre', 'ocupap', 'emprep', 'celularp',
       'telofp', 'emailp', 'pedi', 'telpedi', 'transporta', 'factura',
       'factura1', 'ciclo', 'fechains', 'fechbaja', 'plan', 'apellidos',
       'apellidop', 'apellidom', 'nombres', 'snombre', 'matricula',
       'nombrecort', 'credito', 'cuenta', 'inscribe', 'edad', 'capturado'],
      dtype='object')


### Información y tipos de datos

In [224]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 431 entries, 0 to 430
Data columns (total 52 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   clave       431 non-null    int64         
 1   digito      431 non-null    int64         
 2   digban      431 non-null    int64         
 3   nombrea     431 non-null    object        
 4   sexo        431 non-null    object        
 5   domal       427 non-null    object        
 6   coloa       427 non-null    object        
 7   pobla       424 non-null    object        
 8   cpa         417 non-null    float64       
 9   fecnaca     431 non-null    object        
 10  naciona     428 non-null    object        
 11  tela        399 non-null    object        
 12  nivel       431 non-null    int64         
 13  grado       431 non-null    int64         
 14  grupo       431 non-null    object        
 15  escpro      417 non-null    object        
 16  porcol      431 non-null  

### Primeros 5 registros

In [225]:
df.head()

clave  digito  digban                             nombrea sexo  \
0   3807       0      28               TRUJANO MIRANDA JONAS    M   
1   3752       0      87     SAN NICOLAS BUSTAMANTE SANTIAGO    M   
2   3751       0      27    SAN NICOLAS BUSTAMANTE ESTEFANIA    F   
3   3723       0      46        VIZUET  GOMEZ DIEGO EMILIANO    M   
4   3811       0      25  AGUILAR VAZQUEZ VICTORIA ALEJANDRA    F   

                            domal                 coloa       pobla      cpa  \
0               5 DE MAYO No. 106              ZARAGOZA  TEXCOCO, E  56110.0   
1             LEON SALINAS NO. 62              XOCOTLAN  TEXCOCO. M  56236.0   
2             LEON SALINAS No. 62  FRACC.  MOLINO DE FL  XOCOTLAN,T  56236.0   
3         CIRCUITO MILTENCO No. 4      CABERA MUNICIPAL  CHICOLOAPA  56370.0   
4  FRAY PEDRO DE GANTE No.238 INT             SAN PEDRO     TEXCOCO  56105.0   

               fecnaca  ... apellidom nombres  snombre  matricula nombrecort  \
0  2018-02-12 00:00:00  ...       NaN     NaN      NaN        NaN        NaN   
1  2007-04-19 00:00:00  ...       NaN     NaN      NaN        NaN        NaN   
2  2013-03-27 00:00:00  ...       NaN     NaN      NaN        NaN        NaN   
3  2013-07-15 00:00:00  ...       NaN     NaN      NaN        NaN        NaN   
4  2018-01-12 00:00:00  ...       NaN     NaN      NaN        NaN        NaN   

  credito  cuenta  inscribe  edad capturado  
0     NaN     NaN       NaN     0     False  
1     NaN     NaN       NaN     0     False  
2     NaN     NaN       NaN     0     False  
3     NaN     NaN       NaN     0     False  
4     NaN     NaN       NaN     0     False  

[5 rows x 52 columns]

**Observacion :**  Tenemos varias columnas sin informacion, se hara una revision para poder determinar si eliminarlas o no.

### Valores faltantes

In [226]:
df.isnull().sum()

clave           0
digito          0
digban          0
nombrea         0
sexo            0
domal           4
coloa           4
pobla           7
cpa            14
fecnaca         0
naciona         3
tela           32
nivel           0
grado           0
grupo           0
escpro         14
porcol          0
porins          0
codigo        431
madre           4
ocupam         13
emprem        119
celularm       59
telofm        259
emailm         94
padre          22
ocupap         29
emprep        112
celularp       96
telofp        223
emailp        154
pedi           79
telpedi       115
transporta    431
factura       410
factura1      431
ciclo         431
fechains        0
fechbaja        0
plan            0
apellidos     431
apellidop     431
apellidom     431
nombres       431
snombre       431
matricula     431
nombrecort    431
credito       431
cuenta        431
inscribe      431
edad            0
capturado       0
dtype: int64

**Observacion :**  Teniendo en cuenta que la cantidad de filas del dataframe es de 431 y tenemos columnas con la misma cantidad de valores nulos, podriamos confirmar que son columnas sin informacion, por lo que podrian ser eliminadas.

**Procediemiento :** Se eliminará las columnas donde la cantidad de filas vacias sean igual a la cantidad de filas del dataframe mediante una funcion que irá recorriendo cada columna.

In [227]:
# Determinamos el limite de nulos para eliminar la columna
total_filas = len(df)

# Recorremos cada columna para ir verificando si son columnas vacias
for columna in df.columns:
    # Guardamos la cantidad de nulos en la variable
    nulos_en_columna = df[columna].isnull().sum()
    
    # Verificamos si la cantidad de nulos es igual a la cantidad total de filas de la columna
    if nulos_en_columna == total_filas:
        # Eliminamos columna
        df.drop(columns=columna, inplace=True)
        #Mostramos el mensaje de que columnas fueron eliminadas
        print(f"Columna '{columna}' eliminada (todos los valores son nulos).")

Columna 'codigo' eliminada (todos los valores son nulos).
Columna 'transporta' eliminada (todos los valores son nulos).
Columna 'factura1' eliminada (todos los valores son nulos).
Columna 'ciclo' eliminada (todos los valores son nulos).
Columna 'apellidos' eliminada (todos los valores son nulos).
Columna 'apellidop' eliminada (todos los valores son nulos).
Columna 'apellidom' eliminada (todos los valores son nulos).
Columna 'nombres' eliminada (todos los valores son nulos).
Columna 'snombre' eliminada (todos los valores son nulos).
Columna 'matricula' eliminada (todos los valores son nulos).
Columna 'nombrecort' eliminada (todos los valores son nulos).
Columna 'credito' eliminada (todos los valores son nulos).
Columna 'cuenta' eliminada (todos los valores son nulos).
Columna 'inscribe' eliminada (todos los valores son nulos).


**Resultado**

In [228]:
print(df.columns)

Index(['clave', 'digito', 'digban', 'nombrea', 'sexo', 'domal', 'coloa',
       'pobla', 'cpa', 'fecnaca', 'naciona', 'tela', 'nivel', 'grado', 'grupo',
       'escpro', 'porcol', 'porins', 'madre', 'ocupam', 'emprem', 'celularm',
       'telofm', 'emailm', 'padre', 'ocupap', 'emprep', 'celularp', 'telofp',
       'emailp', 'pedi', 'telpedi', 'factura', 'fechains', 'fechbaja', 'plan',
       'edad', 'capturado'],
      dtype='object')


**Observacion :** Se eliminaron las columnas que no tenian informacion, en el caso de necesitar una nueva columna se podra crear una.

### Duplicados

In [229]:
duplicados = df.duplicated().sum()
print(f"Tenemos {duplicados} registros duplicados.")

Tenemos 0 registros duplicados.


# <font color="#f9a177">**Analisis de columnas**</font>

## <font color="#f9a177">"clave"</font>

### Interpretacion de columna

La columna hace referencia a las claves de cada registro, es decir, cada registro tiene una clave unica.

### Tipo de dato

In [230]:
#Vamos a ver si hay datos distintos de int
df['clave'].apply(lambda x: type(x)).value_counts()

clave
<class 'int'>    431
Name: count, dtype: int64

### Valores faltantes

In [231]:
#Valores igual a "Nan"
df_nan = df[df['clave'].apply(lambda x: x == 'Nan')]

In [232]:
#Valores nulos
nulos = df['clave'].isnull().sum()
#Sumamos nulos y Nan
print(f"Valores faltantes en la columna 'clave': {nulos + len(df_nan)}") 

Valores faltantes en la columna 'clave': 0


### Valores duplicados

In [233]:
df[df['clave'].duplicated(keep=False)]

clave  digito  digban                nombrea sexo                  domal  \
145   3654       0      42  ARIAS CAMPOS SAMANTHA    F  AV. NACIONAL LT. 42 A   
413   3654       0      52      CARREON REYES AMY    F          LAREDO No. 35   

                    coloa       pobla      cpa              fecnaca  ...  \
145  SAN FRANCISCO ACUEXC  ATENCO. ME  56300.0  2017-12-01 00:00:00  ...   
413           EL TEJOCOTE  TEXCOCO,ME  56265.0  2015-12-02 00:00:00  ...   

         telofp emailp                         pedi     telpedi factura  \
145  5959223340    NaN                          NaN         NaN     NaN   
413         NaN    NaN  DR. MANUEL ALEJANDRO CAMPOS  5556851986     NaN   

      fechains  fechbaja  plan edad capturado  
145 2022-03-24     -   -     B    0     False  
413 2022-09-05     -   -     B    0     False  

[2 rows x 38 columns]

**Observacion :**  Se puede observar que tenemos dos claves duplicadas, lo cual probablemente sea un error en la carga de datos ya que cada registro tiene una clave unica, no se puede eliminar estos registros ya que no son duplicados exactos, es evidente que pertenecen a diferentes padres y alumnos. Se realizara un solucion luego de verificar otras columnas.

## <font color="#f9a177">"digito"</font>

### Interpretacion

Podria hacer referencia a una extension de la misma clave, un numero adicional a la clave. Tambien tener en cuenta que estos numeros si se repiten.

### Tipo de dato

In [234]:
#Vamos a ver si hay datos distintos de int
df['digito'].apply(lambda x: type(x)).value_counts()

digito
<class 'int'>    431
Name: count, dtype: int64

### Valores faltantes

In [235]:
#Valores igual a "Nan"
df_nan = df[df['digito'].apply(lambda x: x == 'Nan' or x == 'NaN')]

In [236]:
nulos = df['digito'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'digito': {nulos + len(df_nan)}")

Valores faltantes en la columna 'digito': 0


### Valores duplicados

In [237]:
#Cantidad de duplicados por digito
df['digito'].value_counts()

digito
0    429
1      1
5      1
Name: count, dtype: int64

**Observacion :**  Se puede observar dos registros con digitos distintos, esto puede ser un error en la carga de datos y la mayoria de los registros tiene el digito en 0. Considero que esta columna no tiene informacion relevante, si quisiera crear claves unicas unificando esta columna con la columna "clave" seguiria teniendo el mismo problema de duplicados.

**Procedimiento :** Se eliminará la columna "digito".

In [238]:
df.drop(columns='digito', inplace=True)

**Resultado**

In [239]:
df.head()

clave  digban                             nombrea sexo  \
0   3807      28               TRUJANO MIRANDA JONAS    M   
1   3752      87     SAN NICOLAS BUSTAMANTE SANTIAGO    M   
2   3751      27    SAN NICOLAS BUSTAMANTE ESTEFANIA    F   
3   3723      46        VIZUET  GOMEZ DIEGO EMILIANO    M   
4   3811      25  AGUILAR VAZQUEZ VICTORIA ALEJANDRA    F   

                            domal                 coloa       pobla      cpa  \
0               5 DE MAYO No. 106              ZARAGOZA  TEXCOCO, E  56110.0   
1             LEON SALINAS NO. 62              XOCOTLAN  TEXCOCO. M  56236.0   
2             LEON SALINAS No. 62  FRACC.  MOLINO DE FL  XOCOTLAN,T  56236.0   
3         CIRCUITO MILTENCO No. 4      CABERA MUNICIPAL  CHICOLOAPA  56370.0   
4  FRAY PEDRO DE GANTE No.238 INT             SAN PEDRO     TEXCOCO  56105.0   

               fecnaca   naciona  ...      telofp                    emailp  \
0  2018-02-12 00:00:00  MEXICANA  ...         NaN                       NaN   
1  2007-04-19 00:00:00  MEXICANA  ...  5519882072  rosamaria-rmbo@gmail.com   
2  2013-03-27 00:00:00  MEXICANA  ...  5519882072  rosamaria.rmbo@gmail.com   
3  2013-07-15 00:00:00  MEXICANA  ...         NaN                       NaN   
4  2018-01-12 00:00:00  MEXICANA  ...         NaN       harrohari@gmail.com   

                    pedi     telpedi  \
0   DRA. VERONICA DELADO  5959547699   
1  DRA. ALEJANDRA MUENCH  5959252210   
2  DRA. ALEJANDRA MUENCH  5959252210   
3                    NaN         NaN   
4           DR. GO0NGORA  5959555296   

                                             factura   fechains  fechbaja  \
0                                                NaN 2022-02-11     -   -   
1                                                NaN 2022-02-11     -   -   
2                                                NaN 2022-02-11     -   -   
3  FACTURAR A NOMBRE DE: CATERIN LIZBETH GOMEZ TI... 2022-02-11     -   -   
4                                                NaN 2022-02-14     -   -   

  plan edad capturado  
0    B    0     False  
1    B    0     False  
2    B    0     False  
3    B    0     False  
4    B    0     False  

[5 rows x 37 columns]

## <font color="#f9a177">"digban"</font>

### Interpretacion

Podria ser un registro relacionado con una entidad bancaria o parte del de una tarjeta bancaria.

### Tipo de dato

In [240]:
#Vamos a ver si hay datos distintos de int
df['digban'].apply(lambda x: type(x)).value_counts()

digban
<class 'int'>    431
Name: count, dtype: int64

### Valores faltantes

In [241]:
#Valores igual a "Nan"
df_nan = df[df['digban'].apply(lambda x: x == 'Nan' or x == 'NaN')]

In [242]:
nulos = df['digban'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'digban': {nulos + len(df_nan)}")

Valores faltantes en la columna 'digban': 0


### Valores duplicados

In [243]:
#valores duplicados en digban
df['digban'].value_counts()

digban
73    9
65    9
63    9
22    8
46    8
     ..
18    1
72    1
39    1
14    1
74    1
Name: count, Length: 97, dtype: int64

**Observacion :**  Tenemos numeros duplicados en los registros, podria no ser un problema ya que el numero podria corresponder a la terminacion de una tarjeta bancaria o el numero para una entidad bancaria por lo que tendriamos miles de variantes

### Formato de los datos

In [244]:
#verificar si todos los datos tienen dos digitos y si no tiene mostrar estos datos
df[df['digban'].apply(lambda x: len(str(x)) != 2)][['clave', 'digban']]

clave  digban
37    3339       6
62    3655       5
84    3676       3
86    3433       5
89    2986       5
99    3890       8
108   3620       9
116   3685       9
137   3027       4
151   3898       3
165   3604       3
171   3856       7
176   3766       5
183   3211       5
217   3484       7
220   3502       9
254   3207       8
266   3860       4
268   3787       3
284   3565       3
289   3916       5
296   3241       9
305   3796       9
332   3189       6
361   3937       3
376   3745       7
380   3946       9
383   2632       7
398   3950       6
425   3971       4

**Observacion :**  Se puede observar que tenemos datos con un digito, esto podria tratarse algun numero de referencia a una entidad bancaria, ya que cualquier tarjeta de credito tiene numeros suficientes para poder cargar.

**Procedimiento :** Se unificara la columna "digban" con la columna "clave" para poder tener una clave unica para cada registro y como paso final se volvera a dejar la columna "clave" como tipo de dato int para no tener conflictos

In [245]:
#Vamos a sobrescribir en la columna "clave" la union de las columnas "clave" y "digban"
#Para unir ambas columnas sera necesario convertir a string las columnas y sumarlas
df['clave'] = df['clave'].astype(str) + df['digban'].astype(str)
#Eliminamos la columna "digban"
df.drop(columns='digban', inplace=True)


**Resultado**

In [246]:
df["clave"].head()

0    380728
1    375287
2    375127
3    372346
4    381125
Name: clave, dtype: object

In [247]:
#Cambiar str a int
df['clave'] = df['clave'].astype(int)

### Verificamos duplicados en clave

In [248]:
df[df['clave'].duplicated(keep=False)]

Empty DataFrame
Columns: [clave, nombrea, sexo, domal, coloa, pobla, cpa, fecnaca, naciona, tela, nivel, grado, grupo, escpro, porcol, porins, madre, ocupam, emprem, celularm, telofm, emailm, padre, ocupap, emprep, celularp, telofp, emailp, pedi, telpedi, factura, fechains, fechbaja, plan, edad, capturado]
Index: []

[0 rows x 36 columns]

**Observacion :**  Se puede observar que al unificar estas columnas tendriamos resuelto el problema de las claves duplicadas, ya que cada clave es unica.

## <font color="#f9a177">"porins"</font>

### Interpretacion

Podria referirse al porcentaje de inscripcion

**Procedimiento :** Tratare de entender a que hace referencia esta columna

In [249]:
#valores unicos
df['porcol'].unique()

array([100,  85,  75,   0,  90,  95,  30,   1,  80,  35])

**Observacion :**  Se puede observar que el supuesto porcentaje de inscripcion podria hacer referencia a un porcentaje del total de la inspricion que se pago en el momento registrado. Por las dudas dejare la columna por si pueda ser util para alguna promocion o beneficio

### Tipo de dato

In [250]:
#Vamos a ver si hay datos distintos de int
df['porins'].apply(lambda x: type(x)).value_counts()

porins
<class 'int'>    431
Name: count, dtype: int64

### Valores faltantes

In [251]:
#Valores igual a "Nan"
df_nan = df[df['porcol'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [252]:
nulos = df['porcol'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'porcol': {nulos + len(df_nan)}")

Valores faltantes en la columna 'porcol': 0


## <font color="#f9a177">"factura"</font>

### Interpretacion

Hace referencia a la factura de cada alumno

**Procedimiento :** Tratare de entender a que hace referencia esta columna

In [253]:
#valores unicos
df['factura'].unique()

array([nan, 'FACTURAR A NOMBRE DE: CATERIN LIZBETH GOMEZ TIRADO',
       'FACTURA A NOMBRE DE: CONACORP, S.A. DE C.V.',
       'A NOMBRE DE: ITZEL GALEANA  RANGEL',
       'A NOMBRE DE: ITZEL GALEANA RANGEL',
       'FACTURA A NOMBRE DE: LEON OMAR VALVERDE ROJAS',
       'SRA. MARIA LUISA JIMENEZ HERRERA', 'SR. LUIS NARCISO CANACASCO',
       'A NOMBRE DE RODOLFO HERNANDEZ FLORES',
       'FACTURAR A: DURAN TOVAR ALEJANDRA',
       'FACTURA A NOMBRE DE EDGAR ISAAC RAMIREZ MONTIEL',
       'REQUIERE FACTURA A  NOMBRE DE: EDGAR PEREZ JUAREZ',
       'FACTURAR A NOMBRE DE: CONACORP, S.A. DE C.V.',
       'SAIRI JAZEL CRUZ VELEZ', 'NO REQUIERE DE FACTURA',
       'GOMEZ SANCHEZ OSCAR', 'NAHUM LOPEZ GOMEZ', 'ULISES ZAMORA MEJIA',
       'ZAMORA CHIMAL ULISES', 'TEL. ABUELO 55 36286271',
       'FACRURAR A: ISABEL NOEMI ARIAS HERNANDEZ',
       'BRENDA DEL CARMEN  ARENAS ARENAS'], dtype=object)

In [254]:
#ver columna factura y nomprep y nombrem para encontrar alguna relacion
df[['factura', 'padre', 'madre']]

factura  \
0                                                  NaN   
1                                                  NaN   
2                                                  NaN   
3    FACTURAR A NOMBRE DE: CATERIN LIZBETH GOMEZ TI...   
4                                                  NaN   
..                                                 ...   
426                                                NaN   
427                                                NaN   
428                                                NaN   
429                                                NaN   
430                                                NaN   

                              padre                           madre  
0                               NaN  GLORIA CATALINA TRUJANO MIRAND  
1        CARLOS SAN NICOLAS MIRANDA  ROSA MARIA BUSTAMANTE OSEGUERA  
2     CARLOS SAN NICOLAS BUSTAMANTE  ROSA MARIA BUSTAMANTE OSEGUERA  
3           JOSE PEDRO VIZUET LOPEZ     MARIA DEL CARMEN TIRADO A,.  
4      ALLSON HARI AGUILAR TRUJILLO      DIANA LAURA VAZQUEZ MENDEZ  
..                              ...                             ...  
426  CARLOS ALBERTO VELAZQUEZ GONZA                   ITZEL RAMIREZ  
427            MAURICIO GONZALEZ V.              MARICARMEN ESPEJEL  
428  MARIO ALBERTO CLAVIJO CASTILLO  LIDIA STEPHANIE ELIZALDE TORRE  
429                             NaN                             NaN  
430          SAUL HERNANDEZ VAZQUEZ         MARISOL GUTIERREZ MUÑOZ  

[431 rows x 3 columns]

### Valores faltantes

In [255]:
#Valores igual a "Nan"
df_nan = df[df['factura'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [256]:
nulos = df['factura'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'factura': {nulos + len(df_nan)}")

Valores faltantes en la columna 'factura': 410


**Observacion :**  Se puede observar que la columna hace referencia a quien debe facturar, si el padre o la madre, el problema es que tenemos bastante informacion faltante por lo que termina siendo una columna no tan util, seria importante consultar nuevamente a quien envio los datos para tener informacion clara 

**Procedimiento :** Se eliminara la columna "factura"

In [257]:
df.drop(columns='factura', inplace=True)

## <font color="#f9a177">"fechains"</font>

### Interpretacion

Hace referencia a la fecha de inscripcion del alumno

**Procedimiento :** Cambiare el nombre de la columna a "fecha_inscripcion" para mayor claridad

In [258]:
df.rename(columns={'fechains': 'fecha_inscripcion'}, inplace=True)

**Resultado**

In [259]:
df['fecha_inscripcion'].head()

0   2022-02-11
1   2022-02-11
2   2022-02-11
3   2022-02-11
4   2022-02-14
Name: fecha_inscripcion, dtype: datetime64[ns]

### Tipo de dato

In [260]:
#Vamos a ver si hay datos distintos de datatime
df['fecha_inscripcion'].apply(lambda x: type(x)).value_counts()

fecha_inscripcion
<class 'pandas._libs.tslibs.timestamps.Timestamp'>    431
Name: count, dtype: int64

### Valores faltantes

In [261]:
#Valores igual a "Nan"
df_nan = df[df['fecha_inscripcion'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [262]:
nulos = df['fecha_inscripcion'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'fecha_inscripcion': {nulos + len(df_nan)}")

Valores faltantes en la columna 'fecha_inscripcion': 0


### Formato de datos

In [263]:
#valores unicos
df['fecha_inscripcion'].unique()

<DatetimeArray>
['2022-02-11 00:00:00', '2022-02-14 00:00:00', '2022-02-15 00:00:00',
 '2022-02-16 00:00:00', '2022-02-18 00:00:00', '2022-02-21 00:00:00',
 '2022-02-22 00:00:00', '2022-02-23 00:00:00', '2022-02-24 00:00:00',
 '2022-02-25 00:00:00',
 ...
 '2023-01-30 00:00:00', '2023-02-15 00:00:00', '2023-02-28 00:00:00',
 '2023-03-01 00:00:00', '2023-02-27 00:00:00', '2023-03-15 00:00:00',
 '2023-03-27 00:00:00', '2023-03-29 00:00:00', '2023-04-17 00:00:00',
 '2023-05-02 00:00:00']
Length: 116, dtype: datetime64[ns]

**Procedimiento :**  Se puede observar que los datos de la columna no estan en un formato adecuado, se procedera a cambiar el formato de los datos.

In [264]:
#Transformamos los datos que estan como datetime.datetime(2022, 10, 5, 0, 0) a un formato de fecha adecuado
# Convertimos la columna a datetime especificando el formato
df['fecha_inscripcion'] = pd.to_datetime(df['fecha_inscripcion'], format='%Y-%m-%d', errors='coerce')

# Aplicamos el formato deseado
df['fecha_inscripcion'] = df['fecha_inscripcion'].apply(lambda x: x.strftime('%Y-%m-%d') if pd.notnull(x) else '')

**Resultado**

In [265]:
#Valores unicos
df['fecha_inscripcion'].unique()

array(['2022-02-11', '2022-02-14', '2022-02-15', '2022-02-16',
       '2022-02-18', '2022-02-21', '2022-02-22', '2022-02-23',
       '2022-02-24', '2022-02-25', '2022-02-28', '2022-03-01',
       '2022-03-02', '2022-03-03', '2022-03-04', '2022-03-07',
       '2022-03-08', '2022-03-09', '2022-03-10', '2022-03-11',
       '2022-03-14', '2022-03-15', '2022-03-16', '2022-03-17',
       '2022-03-18', '2022-03-22', '2022-03-23', '2022-03-24',
       '2022-03-25', '2022-03-28', '2022-03-29', '2022-03-30',
       '2022-03-31', '2022-04-01', '2022-04-04', '2022-04-05',
       '2022-04-07', '2022-04-08', '2022-04-13', '2022-04-25',
       '2022-04-27', '2022-04-29', '2022-05-02', '2022-05-03',
       '2022-05-12', '2022-05-17', '2022-05-25', '2022-05-27',
       '2022-05-31', '2022-06-02', '2022-06-06', '2022-06-08',
       '2022-06-10', '2022-06-13', '2022-06-15', '2022-06-16',
       '2022-06-21', '2022-06-22', '2022-06-23', '2022-06-27',
       '2022-06-28', '2022-06-29', '2022-06-30', '2022-

## <font color="#f9a177">"fechbaja"</font>

### Interpretacion

Hace referencia a la fecha de baja del alumno

**Procedimiento :** Cambiare el nombre de la columna a "fecha_baja" para mayor claridad

In [266]:
df.rename(columns={'fechbaja': 'fecha_baja'}, inplace=True)

**Resultado**

In [267]:
df['fecha_baja'].head()

0      -   -
1      -   -
2      -   -
3      -   -
4      -   -
Name: fecha_baja, dtype: object

### Tipo de dato

In [268]:
#Vamos a ver si hay datos distintos de datatime
df['fecha_baja'].apply(lambda x: type(x)).value_counts()

fecha_baja
<class 'str'>                  419
<class 'datetime.datetime'>     12
Name: count, dtype: int64

**Observacion :**  Tenemos datos de un tipo el cual no corresponde al sentido de la columna

### Formato de los datos

In [269]:
#Valores unicos
df['fecha_baja'].unique()

array(['  -   -', datetime.datetime(2023, 1, 9, 0, 0),
       datetime.datetime(2022, 10, 5, 0, 0),
       datetime.datetime(2022, 11, 30, 0, 0),
       datetime.datetime(2022, 7, 6, 0, 0),
       datetime.datetime(2022, 12, 13, 0, 0),
       datetime.datetime(2022, 10, 25, 0, 0),
       datetime.datetime(2022, 7, 22, 0, 0),
       datetime.datetime(2022, 8, 22, 0, 0),
       datetime.datetime(2022, 8, 12, 0, 0),
       datetime.datetime(2022, 9, 13, 0, 0),
       datetime.datetime(2022, 9, 5, 0, 0)], dtype=object)

In [270]:
#Transformamos los datos que estan como datetime.datetime(2022, 10, 5, 0, 0) a un formato de fecha adecuado
# Convertimos la columna a datetime especificando el formato
df['fecha_baja'] = pd.to_datetime(df['fecha_baja'], format='%Y-%m-%d', errors='coerce')

# Aplicamos el formato deseado
df['fecha_baja'] = df['fecha_baja'].apply(lambda x: x.strftime('%Y-%m-%d') if pd.notnull(x) else '')

**Resultado**

In [271]:
#Valores unicos
df['fecha_baja'].unique()

array(['', '2023-01-09', '2022-10-05', '2022-11-30', '2022-07-06',
       '2022-12-13', '2022-10-25', '2022-07-22', '2022-08-22',
       '2022-08-12', '2022-09-13', '2022-09-05'], dtype=object)

In [272]:
df['fecha_baja'].head()

0    
1    
2    
3    
4    
Name: fecha_baja, dtype: object

### Valores faltantes

In [273]:
#Valores igual a "Nan"
df_nan = df[df['fecha_baja'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [274]:
nulos = df['fecha_baja'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'fecha_baja': {nulos + len(df_nan)}")

Valores faltantes en la columna 'fecha_baja': 419


**Observacion :**  Tenemos muchos registros con fechas de baja faltantes, considero que es necesario eliminar esta columna ya que ofrece muy poca informacion.

**Procedimiento :** Se eliminará la columna "fecha_baja"

In [275]:
df.drop(columns='fecha_baja', inplace=True)

## <font color="#f9a177">"plan"</font>

### Interpretacion

Hace referencia al plan de estudios del alumno o plan de pago

### Tipo de dato

In [276]:
#Vamos a ver si hay datos distintos
df['plan'].apply(lambda x: type(x)).value_counts()

plan
<class 'str'>    431
Name: count, dtype: int64

### Valores faltantes

In [277]:
#Valores igual a "Nan"
df_nan = df[df['plan'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [278]:
nulos = df['plan'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'plan': {nulos + len(df_nan)}")

Valores faltantes en la columna 'plan': 0


### Formato de los datos

In [279]:
#Valores unicos
df['plan'].unique()

array(['B', 'A'], dtype=object)

**Observacion :**  Tenemos dos tipos de planes no se bien a que podria hace referencia si al plan de estudio o el plan de pago, por el momento mantendre la columna ya que se podria ser util la infomracion 

## <font color="#f9a177">"capturado"</font>

### Interpretacion

Podria hacer referencia a quien capturo los datos 

**Procedimiento :** Tratare de entender a que hace referencia esta columna

### Tipo de dato

In [280]:
#Vamos a ver si hay datos distintos
df['capturado'].apply(lambda x: type(x)).value_counts()

capturado
<class 'bool'>    431
Name: count, dtype: int64

### Valores faltantes

In [281]:
#Valores igual a "Nan"
df_nan = df[df['capturado'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [282]:
nulos = df['capturado'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'capturado': {nulos + len(df_nan)}")

Valores faltantes en la columna 'capturado': 0


### Formato de los datos

In [283]:
#Valores unicos
df['capturado'].unique()

array([False])

**Observacion :**  Realmente no puedo determinar a qeu puede hacer referencia esta columna, ademas tenemos un solo valor, considero que no es una columna importante para el dataframe

**Procedimiento :** Se eliminará la columna "capturado"

In [284]:
df.drop(columns='capturado', inplace=True)

# <font color="#f9a177">**Analisis de columnas relacionadas al alumno**</font>

## <font color="#f9a177">"sexo"</font>

### Interpretacion

Se refiere al sexo de cada alumno masculino y femenino (M y F).

### Tipo de dato

In [285]:
#Vamos a ver si hay datos distintos de string
df['sexo'].apply(lambda x: type(x)).value_counts()

sexo
<class 'str'>    431
Name: count, dtype: int64

### Valores faltantes

In [286]:
#Valores igual a "Nan"
df_nan = df[df['sexo'].apply(lambda x: x == 'Nan' or x == 'NaN')]

In [287]:
nulos = df['sexo'].isnull().sum()
print(f"Valores faltantes en la columna 'sexo': {nulos + len(df_nan)}")

Valores faltantes en la columna 'sexo': 0


### Valores distintos

In [288]:
#Valores unicos impreso
print(df['sexo'].unique())

['M' 'F']


**Observacion :**  Tenemos dos valores que representan a masculino y femenino.

**Procedimiento :** Se cambiara los valores M y F a Masculino y Femenino respectivamente para una mayor claridad.

In [289]:
#Cambiare M por Masculino y F por Femenino
df['sexo'] = df['sexo'].replace({'M': 'Masculino', 'F': 'Femenino'})

**Resultado**

In [290]:
df["sexo"].head()

0    Masculino
1    Masculino
2     Femenino
3    Masculino
4     Femenino
Name: sexo, dtype: object

## <font color="#f9a177">"domal"</font>

### Interpretacion

Se refiere a la direccion del domicilio del alumno.

**Procedimiento :** Se cambiara el nombre de la columna a "domicilio_alumno" para mayor claridad

In [291]:
df.rename(columns={'domal': 'domicilio_alumno'}, inplace=True)

**Resultado**

In [292]:
df['domicilio_alumno']

0                   5 DE MAYO No. 106
1                 LEON SALINAS NO. 62
2                 LEON SALINAS No. 62
3             CIRCUITO MILTENCO No. 4
4      FRAY PEDRO DE GANTE No.238 INT
                    ...              
426            EMILIANO ZAPATA No. 48
427    CARRIL XOLACHE ADONFO LOPEZ MA
428                      PENSAMIENTOS
429                               NaN
430            MIGUEL HIDALGO  No. 13
Name: domicilio_alumno, Length: 431, dtype: object

### Tipo de dato

In [293]:
#Vamos a ver si hay datos distintos de string
df['domicilio_alumno'].apply(lambda x: type(x)).value_counts()

domicilio_alumno
<class 'str'>      427
<class 'float'>      4
Name: count, dtype: int64

**Procedimiento :**  Tenemos datos de tipo float en la columna, se procedera a cambiar a tipo str para no tener problemas al estandarizar los datos y por la coherenecia de la columna.

In [294]:
df['domicilio_alumno'] = df['domicilio_alumno'].astype(str)

### Valores faltantes

In [295]:
#Valores igual a "Nan"
df_nan = df[df['domicilio_alumno'].apply(lambda x: x == 'Nan' or x == 'NaN')]

In [296]:
nulos = df['domicilio_alumno'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'domicilio_alumno': {nulos + len(df_nan)}")

Valores faltantes en la columna 'domicilio_alumno': 0


**Observacion :**  Tenemos 4 registros sin domicilios, considero que no es necesario eliminar estos registros ya que podrian ser completados en un futuro.

**Procedimiento :** Se completaran los registros nulos con "Domicilio desconocido" para una mayor claridad de los datos faltantes.

In [297]:
#Reemplazamos donde sea nulo por "Domicilio desconocido"
df['domicilio_alumno'] = df['domicilio_alumno'].fillna('Domicilio desconocido')
#Reemplazamos donde sea "Nan" por "Domicilio desconocido"
df['domicilio_alumno'] = df['domicilio_alumno'].replace({'Nan': 'Domicilio desconocido'})

### Estandarizacion de la columna "domicilio_alumno"

**Procedimiento :** Se estandarizaran los datos de la columna "domicilio_alumno" importando la funcion "estandarizacion" del archivo "Estandarizar.py" (En el archivo se encuentra el proceso)

In [298]:
df['domicilio_alumno'] = df['domicilio_alumno'].apply(estandarizacion)

**Resultado**

In [299]:
df['domicilio_alumno']

0                    5 de Mayo No. 106
1                  Leon Salinas No. 62
2                  Leon Salinas No. 62
3              Circuito Miltenco No. 4
4      Fray Pedro de Gante No. 238 Int
                    ...               
426             Emiliano Zapata No. 48
427     Carril Xolache Adonfo Lopez ma
428                       Pensamientos
429                                Nan
430              Miguel Hidalgo No. 13
Name: domicilio_alumno, Length: 431, dtype: object

## <font color="#f9a177">"coloa"</font>

### Interpretacion

Hace referencia a la colonia de cada alumno

**Procedimiento :** Se cambiara el nombre de la columna a "colonia_alumno" para mayor claridad

In [300]:
df.rename(columns={'coloa': 'colonia_alumno'}, inplace=True)

**Resultado**

In [301]:
df['colonia_alumno'].head()

0                ZARAGOZA
1                XOCOTLAN
2    FRACC.  MOLINO DE FL
3        CABERA MUNICIPAL
4               SAN PEDRO
Name: colonia_alumno, dtype: object

### Tipo de dato

In [302]:
#Vamos a ver si hay datos distintos de string
df['colonia_alumno'].apply(lambda x: type(x)).value_counts()

colonia_alumno
<class 'str'>      427
<class 'float'>      4
Name: count, dtype: int64

In [303]:
#ver columnas con valores float
df[df['colonia_alumno'].apply(lambda x: type(x) == float)]

clave                       nombrea       sexo domicilio_alumno  \
129  389454        MARQUEZ  PEREZ  MAXIMO  Masculino              Nan   
346  393417        RODRIGUEZ ACOSTA DARIO  Masculino   Nogales No. 13   
394  394832  CAMACHO TREJO ANTONIO MIGUEL  Masculino              Nan   
429  398010      LEDESMA SANCHEZ SANTIAGO  Masculino              Nan   

    colonia_alumno pobla  cpa              fecnaca   naciona tela  ...  \
129            NaN   NaN  NaN  2018-01-01 00:00:00  MEXICANA  NaN  ...   
346            NaN   NaN  NaN  2017-07-29 00:00:00  MEXICANA  NaN  ...   
394            NaN   NaN  NaN  2018-07-05 00:00:00  MEXICANA  NaN  ...   
429            NaN   NaN  NaN  2019-09-28 00:00:00  MEXICANA  NaN  ...   

      ocupap                emprep      celularp telofp  \
129      NaN                   NaN           NaN    NaN   
346  POLICIA  SRIA.DE SEGURIDAD CI  5.612716e+09    NaN   
394      NaN                   NaN           NaN    NaN   
429      NaN                   NaN           NaN    NaN   

                     emailp                     pedi     telpedi  \
129                     NaN                      NaN         NaN   
346  miguelrdz_r9@hayoo.xom  DRA. ARACELY PERALTA O.  5959541945   
394                     NaN                      NaN         NaN   
429                     NaN                      NaN         NaN   

    fecha_inscripcion plan edad  
129        2022-03-23    B    0  
346        2022-08-10    B    0  
394        2022-08-26    B    0  
429        2023-04-17    B    0  

[4 rows x 33 columns]

**Procedimiento :**  Tenemos datos de tipo float en la columna, se procedera a cambiar a tipo str para no tener problemas al estandarizar los datos y por la coherenecia de la columna.

In [304]:
df['colonia_alumno'] = df['colonia_alumno'].astype(str)

### Valores faltantes

In [305]:
#Valores igual a "Nan"
df_nan = df[df['colonia_alumno'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan')]

In [306]:
#Valores nulos
nulos = df['colonia_alumno'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'colonia_alumno': {nulos + len(df_nan)}")

Valores faltantes en la columna 'colonia_alumno': 4


**Procedimiento :** Se completaran los registros nulos o Nan con "Colonia desconocida" para mayor claridad de los valores faltantes

In [307]:
#Reemplazamos donde sea nulo por "Colonia desconocida"
df['colonia_alumno'] = df['colonia_alumno'].fillna('Colonia desconocida')
#Reemplazamos donde sea Nan por "Colonia desconocida"
df['colonia_alumno'] = df['colonia_alumno'].replace({'Nan': 'Colonia desconocida', 'nan': 'Colonia desconocida', 'NaN': 'Colonia desconocida'})

### Estandarizacion de la columna "colonia_alumno"

**Procedimiento :** Se estandarizaran los datos de la columna "colonia_alumno" importando la funcion "estandarizacion" del archivo "Estandarizar.py" (En el archivo se encuentra el proceso)


In [308]:
df['colonia_alumno'] = df['colonia_alumno'].apply(estandarizacion)

**Resultado**

In [309]:
df['colonia_alumno'].head()

0               Zaragoza
1               Xocotlan
2    Fracc. Molino de fl
3       Cabera Municipal
4              San Pedro
Name: colonia_alumno, dtype: object

## <font color="#f9a177">"pobla"</font>

### Interpretacion

Se refiere a la poblacion de cada alumno

**Procedimiento :** Se cambiara el nombre de la columna a "poblacion_alumno" para mayor claridad

In [310]:
df.rename(columns={'pobla': 'poblacion_alumno'}, inplace=True)

**Resultado**

In [311]:
df['poblacion_alumno'].head()

0    TEXCOCO, E
1    TEXCOCO. M
2    XOCOTLAN,T
3    CHICOLOAPA
4       TEXCOCO
Name: poblacion_alumno, dtype: object

### Tipo de dato

In [312]:
#Vamos a ver si hay datos distintos de string
df['poblacion_alumno'].apply(lambda x: type(x)).value_counts()

poblacion_alumno
<class 'str'>      424
<class 'float'>      7
Name: count, dtype: int64

**Procedimiento :**  Tenemos datos de tipo float en la columna, se procedera a cambiar a tipo str para no tener problemas al estandarizar los datos y por la coherenecia de la columna.

In [313]:
df['poblacion_alumno'] = df['poblacion_alumno'].astype(str)

### Valores faltantes

In [314]:
#Valores igual a "Nan"
df_nan = df[df['poblacion_alumno'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan')]

In [315]:
#Valores nulos
nulos = df['poblacion_alumno'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'poblacion_alumno': {nulos + len(df_nan)}")

Valores faltantes en la columna 'poblacion_alumno': 7


**Procedimiento :** Se completaran los registros nulos o Nan con "Poblacion desconocida" para mayor claridad de los valores faltantes

In [316]:
#Reemplazamos donde sea nulo por "Poblacion desconocida"
df['colonia_alumno'] = df['colonia_alumno'].fillna('Colonia desconocida')
#Reemplazamos donde sea Nan por "Poblacion desconocida"
df['poblacion_alumno'] = df['poblacion_alumno'].replace({'Nan': 'Poblacion desconocida', 'nan': 'Poblacion desconocida', 'NaN': 'Poblacion desconocida'})

### Estadarizacion de la columna "poblacion_alumno"

**Procedimiento :** Se estandarizaran los datos de la columna "domicilio_alumno" importando la funcion "estandarizacion" del archivo "Estandarizar.py" (En el archivo se encuentra el proceso)

In [317]:
df['poblacion_alumno'] = df['poblacion_alumno'].apply(estandarizacion)

**Resultado**

In [318]:
df['poblacion_alumno'].head()

0     Texcoco. e
1     Texcoco. m
2    Xocotlan. t
3     Chicoloapa
4        Texcoco
Name: poblacion_alumno, dtype: object

## <font color="#f9a177">"cpa"</font>

### Interpretacion

Probalemente se refiera al codigo postal del alumno

**Procedimiento :** Se cambiara el nombre de la columna a "codigo_postal_alumno" para mayor claridad

In [319]:
df.rename(columns={'cpa': 'codigo_postal_alumno'}, inplace=True)

**Resultado**

In [320]:
df['codigo_postal_alumno'].head()

0    56110.0
1    56236.0
2    56236.0
3    56370.0
4    56105.0
Name: codigo_postal_alumno, dtype: float64

### Tipo de dato

In [321]:
#Vamos a ver si hay datos distintos de float
df['codigo_postal_alumno'].apply(lambda x: type(x)).value_counts()

codigo_postal_alumno
<class 'float'>    431
Name: count, dtype: int64

**Procedimiento :**  Tenemos datos de tipo float en la columna, se procedera a cambiar a tipo int para no tener problemas al estandarizar los datos y por la coherenecia de la columna. Primero veremos si los datos cumplen el formato

### Formato de los datos

In [322]:
#valores unicos
df['codigo_postal_alumno'].unique()

array([56110., 56236., 56370., 56105., 56130., 56400.,    nan, 56217.,
       56120., 56263., 56203., 54610., 56100., 56235., 56230., 56220.,
       56170., 56377., 56050., 56260., 56270., 56375., 56300., 55486.,
       56140., 56150., 56257., 56215., 55037., 56314., 56247., 56214.,
       26208., 56070., 56033., 56216., 55885.,  8500., 56386., 56373.,
       56246., 55890., 56030., 56243., 56225., 56265., 56020., 56160.,
       56210., 56205., 56085.,  9720., 56335., 56127., 56253., 56227.,
       55884., 57000., 56213., 56347., 56121., 56255., 56272.,  5611.,
       56200., 56273., 56390., 56256., 55870., 56250., 56190., 56108.,
       56234., 15900., 56885., 56208., 56240., 55850., 55763., 56520.,
       56226.,  5620., 56106.])

**Observacion :**  Se puede observar que los codigos son de 5 digitos, hay datos con 4 digitos y otros al parecer con numeros faltantes teniendo en cuenta que en mexico los codigos postales son de 5 digitos. Se ha verificado esto a traves de la siguiente web
https://www.correosdemexico.gob.mx/SSLServicios/ConsultaCP/Descarga.aspx

### Valores faltantes

In [323]:
#Valores igual a "Nan"
df_nan = df[df['codigo_postal_alumno'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan')]

In [324]:
#Valores nulos
nulos = df['codigo_postal_alumno'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'codigo_postal_alumno': {nulos + len(df_nan)}")

Valores faltantes en la columna 'codigo_postal_alumno': 14


**Procedimiento :** Se completaran los registros nulos o Nan con "Codigo postal desconocido" para mayor claridad de los valores faltantes

In [325]:
#Reemplazamos donde sea nulo por "Poblacion desconocida"
df['codigo_postal_alumno'] = df['codigo_postal_alumno'].fillna('Codigo postal desconocido')
#Reemplazamos donde sea Nan por "Poblacion desconocida"
df['codigo_postal_alumno'] = df['codigo_postal_alumno'].replace({'Nan': 'Codigo postal desconocido', 'nan': 'Codigo postal desconocido', 'NaN': 'Codigo postal desconocido'})

**Procedimiento :** Se haran nulos los codigos postales que tengan 4 digitos ya que por un solo digito puede variar un codigo postal, se podra completar consultando a los padres. 

In [326]:
#float a int ignorando 'Codigo postal desconocido'
df['codigo_postal_alumno'] = df['codigo_postal_alumno'].apply(lambda x: int(x) if x != 'Codigo postal desconocido' else x)

In [327]:
#Codigos postales de 4 digitos a Codigo postal desconocido
df['codigo_postal_alumno'] = df['codigo_postal_alumno'].apply(lambda x: 'Codigo postal desconocido' if len(str(x)) < 5 else x)

In [328]:
#valores unicos
df['codigo_postal_alumno'].unique()

array([56110, 56236, 56370, 56105, 56130, 56400,
       'Codigo postal desconocido', 56217, 56120, 56263, 56203, 54610,
       56100, 56235, 56230, 56220, 56170, 56377, 56050, 56260, 56270,
       56375, 56300, 55486, 56140, 56150, 56257, 56215, 55037, 56314,
       56247, 56214, 26208, 56070, 56033, 56216, 55885, 56386, 56373,
       56246, 55890, 56030, 56243, 56225, 56265, 56020, 56160, 56210,
       56205, 56085, 56335, 56127, 56253, 56227, 55884, 57000, 56213,
       56347, 56121, 56255, 56272, 56200, 56273, 56390, 56256, 55870,
       56250, 56190, 56108, 56234, 15900, 56885, 56208, 56240, 55850,
       55763, 56520, 56226, 56106], dtype=object)

**Observacion :**  Se ha cambiado los codigos postales de 4 digitos a "Codigo postal desconocido" para poder completarlos en un futuro.

## <font color="#f9a177">"fecnaca".</font>

### Interpretacion

Se refiere a la fecha de nacimiento de alumno

**Procedimiento :** Se cambiara el nombre de la columna a "fecha_nacimiento_alumno" para mayor claridad.

In [329]:
df.rename(columns={'fecnaca': 'fecha_nacimiento_alumno'}, inplace=True)

**Resultado**

In [330]:
df['fecha_nacimiento_alumno'].head()

0    2018-02-12 00:00:00
1    2007-04-19 00:00:00
2    2013-03-27 00:00:00
3    2013-07-15 00:00:00
4    2018-01-12 00:00:00
Name: fecha_nacimiento_alumno, dtype: object

### Tipo de dato

In [331]:
#Vamos a ver si hay datos distintos de datatime
df['fecha_nacimiento_alumno'].apply(lambda x: type(x)).value_counts()

fecha_nacimiento_alumno
<class 'datetime.datetime'>    430
<class 'str'>                    1
Name: count, dtype: int64

In [332]:
#Motramos el dato donde es de tipo str
df[df['fecha_nacimiento_alumno'].apply(lambda x: type(x) == str)]

clave                     nombrea       sexo    domicilio_alumno  \
105  294992  OLEA ROLDAN DONOVAN DANIEL  Masculino  Corregidora No. 13   

    colonia_alumno poblacion_alumno codigo_postal_alumno  \
105   Tequisistlan         Tezoyuca                56020   

    fecha_nacimiento_alumno   naciona        tela  ...                ocupap  \
105                   -   -  MEXICANA  5959571334  ...  MEDICO VETERINARIO Z   

                   emprep      celularp telofp                emailp  \
105  INSTIT.SALUD EDO. ME  5.524623e+09    NaN  athetl0222@gmail.com   

                         pedi  telpedi fecha_inscripcion plan edad  
105  ATZIN ITAI OLEA GONZALEZ  9571334        2022-03-17    B    0  

[1 rows x 33 columns]

**Observacion :**  Tenemos datos de tipo str en la columnaa lo cual no tendria sentido con la columna de fecha , se va a trabajar el formato y este dato a continuacion.

### Formato de los datos

**Procedimiento :**  Vamos a los datos a 'fecha desconocida' si no cumplen el formato de fecha 

In [333]:
#Eliminamos las hora minutos y segundos
df['fecha_nacimiento_alumno'] = pd.to_datetime(df['fecha_nacimiento_alumno'], errors='coerce')

**Resultado**

In [334]:
df['fecha_nacimiento_alumno'].head()

0   2018-02-12
1   2007-04-19
2   2013-03-27
3   2013-07-15
4   2018-01-12
Name: fecha_nacimiento_alumno, dtype: datetime64[ns]

### Valores faltantes

In [335]:
#Valores igual a "Nan"
df_nan = df[df['fecha_nacimiento_alumno'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan')]

In [336]:
#Valores nulos
nulos = df['fecha_nacimiento_alumno'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'fecha_nacimiento_alumno': {nulos + len(df_nan)}")

Valores faltantes en la columna 'fecha_nacimiento_alumno': 1


**Procedimiento :** Se completaran los registros nulos o Nan con "Fecha desconocida" para mayor claridad de los valores faltantes

In [337]:
df['fecha_nacimiento_alumno'] = df['fecha_nacimiento_alumno'].fillna('Fecha desconocida') 

## <font color="#f9a177">"naciona"</font>

### Interpretacion

Se refiere a la nacionalidad del alumno

**Procedimiento :** Se cambiara el nombre de la columna a "nacionalidad" para mayor claridad.

In [338]:
df.rename(columns={'naciona': 'nacionalidad'}, inplace=True)

**Resultado**

In [339]:
df['nacionalidad'].head()

0    MEXICANA
1    MEXICANA
2    MEXICANA
3    MEXICANA
4    MEXICANA
Name: nacionalidad, dtype: object

### Tipo de dato

In [340]:
#Vamos a ver si hay datos distintos de str
df['nacionalidad'].apply(lambda x: type(x)).value_counts()

nacionalidad
<class 'str'>      428
<class 'float'>      3
Name: count, dtype: int64

**Procedimiento :**  Tenemos datos de tipo float en la columna, se procedera a cambiar a tipo str para no tener problemas al estandarizar los datos y por la coherenecia de la columna.

In [341]:
df['nacionalidad'] = df['nacionalidad'].astype(str)

### Valores faltantes

In [342]:
#Valores igual a "Nan"
df_nan = df[df['nacionalidad'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan')]
#Valores nulos

In [343]:
nulos = df['nacionalidad'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'nacionalidad': {nulos + len(df_nan)}")

Valores faltantes en la columna 'nacionalidad': 3


In [344]:
#Vamos a ver las nacinalidades donde sean nulas y comparar con el codigo postal y el nombre del alumno
df[df['nacionalidad'].apply(lambda x: x == 'nan')][['nacionalidad', 'codigo_postal_alumno', 'nombrea']]


nacionalidad codigo_postal_alumno                   nombrea
84           nan                56243        GARCIA ESTRADA ZOE
103          nan                56100  ROSIQUE ROJAS STIAN AXEL
267          nan                56250   ARREGIN OCAMPO SANTIAGO

**Procediemiento :**  Teniendo en cuenta el nombre del alumno y el codigo de postal cual es comun en Mexico se podria inferir que la nacionalidad es mexicana, se procedera a completar estos valores.

In [345]:
df['nacionalidad'] = df['nacionalidad'].replace({'nan': 'Mexicana'})

### Formato de datos

In [346]:
#Valores diferentes en nacionalidad
df['nacionalidad'].unique()

array(['MEXICANA', 'MEXICANO', 'MEXICA NA', 'MEXCIANA', 'MEXCINA',
       'Mexicana', 'MEXICANAQ', 'MEXICQNA', 'MEXCANA', 'MEXCCIANA',
       'MEXICNA', 'MEXIANA', 'MEXICO', 'MEXICAN A', 'CHINO', 'MEXICANA |',
       'MEXICANA/H', 'M4EXICANA', 'MEXICANA/E'], dtype=object)

**Observacion :**  Tenemos nacionalidades con errores de tipeo, se procedera a corregirlos.

**Procedimiento :** Se corregiran los errores de tipeo en las nacionalidades teniendo en cuenta el sexo del alumno y se estandarizaran los datos.

In [347]:
#Reemplazaremos valores por 'Mexicana' si el dato en la columna sexo es 'Femenino o 'Mexicano' si el dato de la columna sexo es 'Masculino' y si el dato de la columna nacionalidad comienza con el caracter 'M' o 'm'
df['nacionalidad'] = df.apply(lambda x: 'Mexicana' if x['sexo'] == 'Femenino' and x['nacionalidad'][0] == 'M' or x['nacionalidad'][0] == 'm' else x['nacionalidad'], axis=1)
df['nacionalidad'] = df.apply(lambda x: 'Mexicano' if x['sexo'] == 'Masculino' and x['nacionalidad'][0] == 'M' or x['nacionalidad'][0] == 'm' else x['nacionalidad'], axis=1)

**Resultado**

In [348]:
df[['sexo', 'nacionalidad']]

sexo nacionalidad
0    Masculino     Mexicano
1    Masculino     Mexicano
2     Femenino     Mexicana
3    Masculino     Mexicano
4     Femenino     Mexicana
..         ...          ...
426  Masculino     Mexicano
427  Masculino     Mexicano
428  Masculino     Mexicano
429  Masculino     Mexicano
430   Femenino     Mexicana

[431 rows x 2 columns]

#

### Estandarizacion de la columna "nacionalidad"

In [349]:
df['nacionalidad'] = df['nacionalidad'].apply(estandarizacion)

**Resultado**

In [350]:
df['nacionalidad'].unique()

array(['Mexicano', 'Mexicana', 'Chino'], dtype=object)

## <font color="#f9a177">"tela"</font>

### Interpretacion

Se refiere al telefono del alumno

**Procedimiento :** Se cambiara el nombre de la columna a "telefono_alumno" para mayor claridad.

In [351]:
df.rename(columns={'tela': 'telefono_alumno'}, inplace=True)

**Resultado**

In [352]:
df['telefono_alumno'].head()

0    5959541519
1    5959543393
2    5959543393
3    5510491891
4    5956888439
Name: telefono_alumno, dtype: object

### Tipo de datos

In [353]:
#Vamos a ver si hay datos distintos de int
df['telefono_alumno'].apply(lambda x: type(x)).value_counts()

telefono_alumno
<class 'str'>      399
<class 'float'>     32
Name: count, dtype: int64

**Procedimiento :**  Tenemos datos de tipo float en la columna, se procedera a cambiar a tipo int para no tener problemas al estandarizar los datos y por la coherenecia de la columna.

In [354]:
#vemos columnas con valores float
df[df['telefono_alumno'].apply(lambda x: type(x) == float)][['telefono_alumno']]

telefono_alumno
15              NaN
44              NaN
80              NaN
99              NaN
114             NaN
115             NaN
129             NaN
131             NaN
141             NaN
148             NaN
149             NaN
256             NaN
258             NaN
270             NaN
273             NaN
274             NaN
275             NaN
276             NaN
289             NaN
312             NaN
325             NaN
346             NaN
366             NaN
385             NaN
387             NaN
394             NaN
410             NaN
412             NaN
413             NaN
421             NaN
422             NaN
429             NaN

In [355]:
#Valores unicos
df['telefono_alumno'].unique()

array(['5959541519', '5959543393', '5510491891', '5956888439',
       '5541426145', '5534309942', '5533978032', '5545934117',
       '5959547881', '59510415676', '5959310310', '5956887219', nan,
       '9545367', '55 32484708', '5959549592', '5959552952', '5959547440',
       '5959545775', '5544904167', '5531884833', '5530376593',
       '5959549241', '5555015348', '5539559160', '5951491255',
       '5547899051', '5532257809', '5526895806', '5548669116',
       '5956887391', '595 9211242', '5959556381', '5544553309',
       '5618017008', '5610087008', '5949566107', '5537111657',
       '5951087507', '5959536959', '5959523487', '5959557637',
       '5959231168', '5539791730', '5569687735', '5564235636',
       '5959533825', '5959541554', '5951073804', '5949571305',
       '5539313537', '92 50295', '5959531579', '5578130977', '5589071188',
       '5584071188', '5514857005', '5959286068', '5959230422',
       '5566964478', '5959570829', '5959541030', '5580324194',
       '5941082706', '56

**Observacion :**  Primero tener en cuenta que el formato numerico es de 10 digitos, tomando como referencia esto, podemos observar datos que no cumplen este criterio ademas de estar desordenado debido a caracteres especiales y espacios que no corresponden. Tambien mencionar que los datos float son valores faltantes en la columna

### Valores faltantes

In [356]:
#Valores igual a "Nan"
df_nan = df[df['telefono_alumno'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]

In [357]:
#Valores nulos
nulos = df['telefono_alumno'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'telefono_alumno': {nulos + len(df_nan)}")

Valores faltantes en la columna 'telefono_alumno': 32


**Procedimiento :** Se completaran los registros nulos o Nan con "Telefono desconocido" para mayor claridad de los valores faltantes

In [358]:
#Reemplazamos donde sea nulo por "Telefono desconocido"
df['telefono_alumno'] = df['telefono_alumno'].fillna('Telefono desconocido')
#Reemplazamos donde sea Nan por "Telefono Desconocido"
df['telefono_alumno'] = df['telefono_alumno'].replace({'Nan': 'Telefono desconocido', 'nan': 'Telefono desconocido', 'NaN': 'Telefono desconocido', '': 'Telefono desconocido'})

### Formato de los datos

**Procedimiento :**  Se eliminara los caracteres especiales y espacios en blanco de la columna "telefono_alumno" para poder estandarizar los datos con la funcion importada

In [359]:
#Remplazar por 'Telefono desconocido' si el dato tiene menos de 10 digitos
df['telefono_alumno'] = df['telefono_alumno'].apply(lambda x: 'Telefono desconocido' if len(str(x)) < 10 else x)

In [360]:
#Estandarizamos
df['telefono_alumno'] = df['telefono_alumno'].apply(estandarizacion)

In [361]:
#Eliminamos espacios entre numeros(expecto de 'Numero Desconocido'), este proceso se hace para esta columna ya que la funcion estandarizcion es mas conveniente en datos de letras
df['telefono_alumno'] = df['telefono_alumno'].apply(limpiar_numero)

In [362]:
#Tranformamos a int los datos que son distinto de 'Telefono Desconocido'
df['telefono_alumno'] = df['telefono_alumno'].apply(lambda x: int(x) if x != 'Telefono Desconocido' else x)

**Resultado**

In [146]:
df['telefono_alumno'].head()

0    5959541519
1    5959543393
2    5959543393
3    5510491891
4    5956888439
Name: telefono_alumno, dtype: object

**Procedimiento :** Vamos a mofidicar aquellos numeros con mas de 10 digitos

In [147]:
#Vamos a ver los numeros de mas de 10 digitos y si son int
df[df['telefono_alumno'].apply(lambda x: len(str(x)) > 10 and type(x) == int)][['telefono_alumno']]

telefono_alumno
12      59510415676
191     55959544427
232  95510969314452
358  92182389218262

**Procedimiento  :** Para el registro 12 se va eliminar el ultimo digito ya que estaria cumpliendo el estandar con respecto a los demas numeros. Para el registro 191 podemo eliminar el ultimo digito y tendriamos un numero que cumple el estandar. Para el registro 232 podemos eliminar algunos digitos en los extremos y tendriamos un numero adecuado al estandar. Para el registro 358 no veo alguna posible solucion al numero mas que dejarlo en 'Telefono Desconocido'.

In [148]:
#Vamos a eliminar el ultimo numero del registro 12 
df.loc[12, 'telefono_alumno'] = int(str(df.loc[12, 'telefono_alumno'])[:-1])

In [149]:
df['telefono_alumno'][12]

5951041567

In [150]:
#Vamos a eliminar el primer numero del registro 191
df.loc[191, 'telefono_alumno'] = int(str(df.loc[191, 'telefono_alumno'])[1:])

In [151]:
df['telefono_alumno'][191]

5959544427

In [152]:
#Vamos a retener de 95510969314452 el numero 5510969314 del registro 232
df.loc[232, 'telefono_alumno'] = 5510969314

In [153]:
df['telefono_alumno'][232]

5510969314

In [154]:
#Convertimos el registro 358 a 'Telefono Desconocido'
df.loc[358, 'telefono_alumno'] = 'Telefono Desconocido'

In [155]:
df['telefono_alumno'][358]

'Telefono Desconocido'

## <font color="#f9a177">"nivel"</font>

### Interpretacion

Hace referencia al nivel de estudios del alumno

**Procedimiento :** Se cambiara el nombre de la columna a "nivel_estudios" para mayor claridad.

In [156]:
df.rename(columns={'nivel': 'nivel_estudios'}, inplace=True)

**Resultado**

In [157]:
df['nivel_estudios'].head()

0    0
1    2
2    1
3    1
4    0
Name: nivel_estudios, dtype: int64

### Tipo de dato

In [158]:
#Vamos a ver si hay datos distintos de string
df['nivel_estudios'].apply(lambda x: type(x)).value_counts()

nivel_estudios
<class 'int'>    431
Name: count, dtype: int64

### Valores faltantes

In [159]:
#Valores igual a "Nan"
df_nan = df[df['nivel_estudios'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]

In [160]:
#Valores nulos
nulos = df['nivel_estudios'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'nivel_estudios': {nulos + len(df_nan)}")

Valores faltantes en la columna 'nivel_estudios': 0


### Formato de datos

In [161]:
#Valores unicos
df['nivel_estudios'].unique()

array([0, 2, 1])

**Observacion :**  Tenemos 0,1 y 2, segun mi observacion, podrian referirse a los niveles Prescolar, Primaria y Secundaria respectivamente. Se va a verificar luego de completar la columna edad

## <font color="#f9a177">"grado"</font>

### Interpretacion

Hace referencia al grado de estudio del alumno

**Procedimiento :** Se cambiara el nombre de la columna a "grado_estudios" para mayor claridad.

In [162]:
df.rename(columns={'grado': 'grado_estudios'}, inplace=True)

**Resultado**

In [163]:
df['grado_estudios'].head()

0    2
1    3
2    4
3    4
4    2
Name: grado_estudios, dtype: int64

### Tipo de dato

In [164]:
#Vamos a ver si hay datos distintos de int
df['grado_estudios'].apply(lambda x: type(x)).value_counts()

grado_estudios
<class 'int'>    431
Name: count, dtype: int64

### Valores faltantes

In [165]:
#Valores igual a "Nan"
df_nan = df[df['grado_estudios'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]

In [166]:
#Valores nulos
nulos = df['grado_estudios'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'grado_estudios': {nulos + len(df_nan)}")

Valores faltantes en la columna 'grado_estudios': 0


### Formato de datos

In [167]:
#Valores unicos
df['grado_estudios'].unique()

array([2, 3, 4, 1, 5, 6, 0])

**Procemiento :**  Para mejor claridad de esta columna el nivel de grado de estudios comenzara desde 1 en adelante, se procedera a cambiar los valores de 0 a 6 por 1 a 7 respectivamente.

In [168]:
df['grado_estudios'] = df['grado_estudios'].replace({0: 1, 1: 2, 2: 3, 3: 4, 4: 5, 5: 6, 6: 7})

**Resultado**

In [169]:
df['grado_estudios'].head()

0    3
1    4
2    5
3    5
4    3
Name: grado_estudios, dtype: int64

**Procemiento :**  Se va a analizar cada nivel de estudio para ver cuantos grados de estudio tiene con el fin de poder determinar a que niveles de estudio se refiere los numeros de esta columna , tambien teniendo en cuenta informacion solicitada en la web sobre la formacion escolar mexicana.

In [170]:
#Veremos en nivel de estudios 0 cuantos grados de estudio tiene y ordenamos de menor a mayor
df[df['nivel_estudios'] == 0]['grado_estudios'].value_counts().sort_index()

grado_estudios
1    18
2    17
3    21
4    26
Name: count, dtype: int64

In [171]:
#Ver en nivel de estudios 1 cuantos grados de estudio tiene
df[df['nivel_estudios'] == 1]['grado_estudios'].value_counts().sort_index()

grado_estudios
2    50
3    24
4    32
5    42
6    33
7    41
Name: count, dtype: int64

In [172]:
#Ver en nivel de estudios 2 cuantos grados de estudio tiene
df[df['nivel_estudios'] == 2]['grado_estudios'].value_counts().sort_index()

grado_estudios
2    28
3    43
4    56
Name: count, dtype: int64

**Observacion :**  En general la etapa prescolar cuenta con 3 grados, la etapa escolar con 6 grados y la secundaria con 3 grados. Pol lo tanto podemos deducir que el nivel 2 puede corresponder a la etapa prescolar, mientras el nivel 1 a la etapa primaria y finalmente el nivel 3 a la etapa secundaria.

__Fuente :__ https://profelandia.com/sep-divide-en-cuatro-etapas-la-educacion-basica/#:~:text=%C2%ABLa%20educaci%C3%B3n%20b%C3%A1sica%20abarca%20la%20formaci%C3%B3n%20escolar%20de,de%20educaci%C3%B3n%20secundaria%C2%BB%2C%20precisa%20el%20plan%20de%20estudios.

**Procedimiento :**  Se procedera a cambiar los valores de la columna "nivel_estudios" por los valores de la etapa educativa correspondiente.

In [173]:
df['nivel_estudios'] = df['nivel_estudios'].replace({0: 'Prescolar', 1: 'Primaria', 2: 'Secundaria'})

**Resultado**

In [174]:
df['nivel_estudios'].head()

0     Prescolar
1    Secundaria
2      Primaria
3      Primaria
4     Prescolar
Name: nivel_estudios, dtype: object

## <font color="#f9a177">"grupo"</font>

### Interpretacion

Podria referirse al grupo escolar al que pertenece el alumno

In [175]:
#Valores unicos
df['grupo'].unique()

array(['A'], dtype=object)

**Observacion :**  Considero que esta columna teniendo solo un valor, no proporciona informacion relevante al analisis

**Procedimiento :** Se eliminara la columna "grupo"

In [176]:
df.drop(columns='grupo', inplace=True)

## <font color="#f9a177">"escpro"</font>

### Interpretacion

Podria referirse a la escuela a la que pertenece el alumno

**Procedimiento :** Se cambiara el nombre de la columna a "nacionalidad" para mayor claridad.

In [177]:
df.rename(columns={'escpro': 'escuela'}, inplace=True)

**Resultado**

In [178]:
df['escuela'].head()

0    IPANTI
1    IPANTI
2    IPANTI
3    IPANTI
4    IPANTI
Name: escuela, dtype: object

### Tipo de dato

In [179]:
#Vamos a ver si hay datos distintos de str
df['escuela'].apply(lambda x: type(x)).value_counts()

escuela
<class 'str'>      417
<class 'float'>     14
Name: count, dtype: int64

**Procedimiento :**  Tenemos datos de tipo float en la columna, se procedera a cambiar a tipo str para no tener problemas al estandarizar los datos y por la coherenecia de la columna.

In [180]:
df['escuela'] = df['escuela'].astype(str)

### Valores faltantes

In [181]:
#Valores igual a "Nan"
df_nan = df[df['escuela'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [182]:
nulos = df['escuela'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'escuela': {nulos + len(df_nan)}")

Valores faltantes en la columna 'escuela': 14


**Procedimiento :** Se completaran los registros nulos o Nan con "Escuela Desconocida" para mayor claridad de los valores faltantes

In [183]:
#Vamos a ver las escuelas donde sean nulas reemplazar por 'Escuela Desconocida'
#Reemplazamos donde sea nulo por "Escuela Desconocida"
df['escuela'] = df['escuela'].fillna('Escuela Desconocida')
#Reemplazamos donde sea Nan por "Escuela Desconocida"
df['escuela'] = df['escuela'].replace({'Nan': 'Escuela Desconocida', 'nan': 'Escuela Desconocida', 'NaN': 'Escuela Desconocida'})

### Formato de datos

In [184]:
#Valores diferentes en escuelas
df['escuela'].unique()

array(['IPANTI', 'J.NIÑOZ NEXAHUALCOYO', 'TOKANI', 'NUEVO INGRESO',
       'INSTITUTO TEXCOCO', 'Escuela Desconocida', 'KIMDERGYM',
       'KINDERGYM', 'NUEVO INDRESO', 'SEC.GRAL. TEXCOCO 20',
       'COLEGIO MEXICO AMERI', 'LICEO', 'CYNTHIA CUMMINGHAM E',
       'CYNTHIA CUNNINGHAM E', 'PAULO FREIRE', 'MATERNALITOS', 'KINDEGYM',
       'CHINA', 'CENTRO ESCOLAR MOREL', 'KiNDEGYM', 'EURO TEXCOCO',
       'KUNDERGYM', 'KINDER MANUEL DOBLAD', 'CEDRAL',
       'FRANCISCO GABILONDO', 'PRIMARIA URRSS', 'COLEGIO EURO TEXCOCO',
       'PANAMERICANO', 'COLEGIO MADRE DEL DI', 'IBIME',
       'COLEGIO PANAMERICANO', 'ACADEMIA MADOX', 'ENRIQUE REBSAMEN',
       'ARCOIRIS', 'COCUXTA', 'PROYECTO ALANA', 'LEONA VICARIO',
       'COLEGIO EURO', 'EURO'], dtype=object)

**Procedimiento :** Se estandarizara los datos de la columna "escuela" importando la funcion "estandarizacion" del archivo "Estandarizar.py" (En el archivo se encuentra el proceso)

In [185]:
df['escuela'] = df['escuela'].apply(estandarizacion)

**Resultado**

In [186]:
df['escuela'].head()

0    Ipanti
1    Ipanti
2    Ipanti
3    Ipanti
4    Ipanti
Name: escuela, dtype: object

## <font color="#f9a177">"porcol"</font>

### Interpretacion

Podria referirse al porcentaje de colegiatura, tal vez el rendimiento escolar del alumno o su avance respecto al ciclo educativo

**Procedimiento :** Tratare de entender a que hace referencia esta columna

In [187]:
#valores unicos
df['porcol'].unique()

array([100,  85,  75,   0,  90,  95,  30,   1,  80,  35])

In [188]:
#ver donde valor porcol es 100 que datos tiene para nivel de estudios y grado de estudios
df[df['porcol'] == 100][['nivel_estudios', 'grado_estudios']]

nivel_estudios  grado_estudios
0        Prescolar               3
1       Secundaria               4
2         Primaria               5
3         Primaria               5
4        Prescolar               3
..             ...             ...
426     Secundaria               3
427     Secundaria               3
428      Prescolar               1
429      Prescolar               2
430      Prescolar               1

[387 rows x 2 columns]

**Observacion :**  Se puede observar que el porcentaje de colegiatura es 100 para los alumnos de secundaria, por lo tanto se podria inferir que el porcentaje de colegiatura es el avance del alumno en el ciclo escolar pero podemos observar que no se encuentran en el ultimo grado lo cual refuta mi hipotesis. La otra teoria que tengo es que puede hacer referencia a l rendimiento escolar del alumno, si fuera el caso dejaria la columna ya que un alumno con buen porcentaje de notas podria recibir alguna promocion o beca

### Tipo de dato

In [189]:
#Vamos a ver si hay datos distintos de int
df['porcol'].apply(lambda x: type(x)).value_counts()

porcol
<class 'int'>    431
Name: count, dtype: int64

### Valores faltantes

In [190]:
#Valores igual a "Nan"
df_nan = df[df['porcol'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [191]:
nulos = df['porcol'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'porcol': {nulos + len(df_nan)}")

Valores faltantes en la columna 'porcol': 0


## <font color="#f9a177">"edad"</font>

### Interpretacion

Hace referencia a la edad del alumno

**Procedimiento :** Se cambiara el nombre de la columna a "edad_alumno" para mayor claridad.

In [192]:
df.rename(columns={'edad': 'edad_alumno'}, inplace=True)

### Tipo de dato

In [193]:
#Vamos a ver si hay datos distintos
df['edad_alumno'].apply(lambda x: type(x)).value_counts()

edad_alumno
<class 'int'>    431
Name: count, dtype: int64

### Valores faltantes

In [194]:
#Valores igual a "Nan"
df_nan = df[df['edad_alumno'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [195]:
nulos = df['edad_alumno'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'edad_alumno': {nulos + len(df_nan)}")

Valores faltantes en la columna 'edad_alumno': 0


### Formato de datos

In [196]:
#Valores unicos
df['edad_alumno'].unique()

array([0])

**Procedimiento :**  Se va a completar la edad de los alumnos tomando la fecha actual con la fecha de nacimiento del alumno

In [197]:
#en columna edad_alumno restar el año actual 2024 con el año de nacimiento si la fecha de nacimiento es distintia de 'Fecha Desconocida'
df['edad_alumno'] = df.apply(lambda x: 2024 - x['fecha_nacimiento_alumno'].year if x['fecha_nacimiento_alumno'] != 'Fecha desconocida' else x['edad_alumno'], axis=1)

In [198]:
#en columna edad_alumno vamos a remplazar por 'Edad Desconocida' si el año de nacimiento es 'Fecha Desconocida'
df['edad_alumno'] = df.apply(lambda x: 'Edad Desconocida' if x['fecha_nacimiento_alumno'] == 'Fecha desconocida' else x['edad_alumno'], axis=1)

In [199]:
#en columna edad_alumno vamos a remplazar por 'Edad Desconocida' si el año de nacimiento es igual o mayor a 2024 para evitar edades negativas 
#(Por seguridad transformamos a datatime para poder tomar el año para comparar
df['fecha_nacimiento_alumno'] = pd.to_datetime(df['fecha_nacimiento_alumno'], errors='coerce')
df['edad_alumno'] = df.apply(lambda x: 'Edad Desconocida' if x['fecha_nacimiento_alumno'].year >= 2024 else x['edad_alumno'], axis=1)

**Resultado**

In [200]:
df['edad_alumno']

0       6
1      17
2      11
3      11
4       6
       ..
426    15
427    15
428     4
429     5
430     3
Name: edad_alumno, Length: 431, dtype: object

In [201]:
#valores unicos
df['edad_alumno'].unique()

array([6, 17, 11, 9, 16, 8, 3, 12, 13, 10, 15, 14, 2, 7, 5, 4, 23,
       'Edad Desconocida'], dtype=object)

## <font color="#f9a177">"nombrea"</font>

### Interpretacion

Hace referencia al nombre del alumno

**Procedimiento :** Se cambiara el nombre de la columna a "nombre alumno" para mayor claridad.

In [202]:
df.rename(columns={'nombrea': 'nombre_alumno'}, inplace=True)

**Resultado**

In [203]:
df['nombre_alumno'].head()

0                 TRUJANO MIRANDA JONAS
1       SAN NICOLAS BUSTAMANTE SANTIAGO
2      SAN NICOLAS BUSTAMANTE ESTEFANIA
3          VIZUET  GOMEZ DIEGO EMILIANO
4    AGUILAR VAZQUEZ VICTORIA ALEJANDRA
Name: nombre_alumno, dtype: object

### Tipo de dato

In [204]:
#Vamos a ver si hay datos distintos de str
df['nombre_alumno'].apply(lambda x: type(x)).value_counts()

nombre_alumno
<class 'str'>    431
Name: count, dtype: int64

### Valores faltantes

In [205]:
#Valores igual a "Nan"
df_nan = df[df['nombre_alumno'].apply(lambda x: x == 'NaN' or x == 'Nan' or x == 'nan' or x == '')]
#Valores nulos

In [206]:
nulos = df['nombre_alumno'].isnull().sum()
#Suma de nulos y Nan
print(f"Valores faltantes en la columna 'nombre_alumno': {nulos + len(df_nan)}")

Valores faltantes en la columna 'nombre_alumno': 0


### Formato de datos

In [207]:
#Valores diferentes en escuelas
df['nombre_alumno'].unique()

array(['TRUJANO MIRANDA JONAS', 'SAN NICOLAS BUSTAMANTE SANTIAGO',
       'SAN NICOLAS BUSTAMANTE ESTEFANIA', 'VIZUET  GOMEZ DIEGO EMILIANO',
       'AGUILAR VAZQUEZ VICTORIA ALEJANDRA', 'UEDA HERNANDEZ LIA SINAI',
       'UEDA HERNaNDEZ MATIAS AMIR', 'RUIZ LEYVA JIMENA',
       'MONTOYA  ZAVALA ANA SOFIA', 'EVANGELISTA CARRERA MAXIMILIANO',
       'MONTOYA ZAVALA DIEGO KALED', 'LOPEZ SANCHEZ ALYNE RENATA',
       'PIÑA CERVANTES JEROME EDUARDO', 'GALLEGOS LOPEZ DAYRON AMAURY',
       'FLORES SORIANO RENATA MAYTE', 'FLORES SORIANO XIMENA MONSERRAT',
       'OROZCO MENDOZA SANTIAGO AYEL', 'FONSECA CANTU LAURA ISABEL',
       'PARDO LEON ALISON TAMARA', 'GODOY ALONSO RAFAEL',
       'PEREZ HERNANDEZ IKER DANIEL', 'CRUZ MARTINEZ ABELARDO GAEL',
       'ROCHA LOPEZ DARA SOFIA', 'ALVAREZ GARCIA MILA',
       'AMEZCUA LARQUE FRANCISCO CARLOS', 'ESPINOZA OLIVARES LUIS SAMUEL',
       'PEREZ SANCHEZ LEONARDO DAVID', 'MEDINA DE LA CRUZ AXEL SAID',
       'MEDINA DE LA CRUZ HIRAM KARIM', 'LEMOS 

**Procedimiento :** Se estandarizara los datos de la columna "nombrea_alumno" importando la funcion "estandarizacion" del archivo "Estandarizar.py" (En el archivo se encuentra el proceso)

In [208]:
df['nombre_alumno'] = df['nombre_alumno'].apply(estandarizacion)

**Resultado**

In [209]:
df['nombre_alumno'].head()

0                 Trujano Miranda Jonas
1       San Nicolas Bustamante Santiago
2      San Nicolas Bustamante Estefania
3           Vizuet Gomez Diego Emiliano
4    Aguilar Vazquez Victoria Alejandra
Name: nombre_alumno, dtype: object

**Resultado**

In [210]:
df['nombre_alumno'].unique()

array(['Trujano Miranda Jonas', 'San Nicolas Bustamante Santiago',
       'San Nicolas Bustamante Estefania', 'Vizuet Gomez Diego Emiliano',
       'Aguilar Vazquez Victoria Alejandra', 'Ueda Hernandez Lia Sinai',
       'Ueda Hernandez Matias Amir', 'Ruiz Leyva Jimena',
       'Montoya Zavala Ana Sofia', 'Evangelista Carrera Maximiliano',
       'Montoya Zavala Diego Kaled', 'Lopez Sanchez Alyne Renata',
       'Piña Cervantes Jerome Eduardo', 'Gallegos Lopez Dayron Amaury',
       'Flores Soriano Renata Mayte', 'Flores Soriano Ximena Monserrat',
       'Orozco Mendoza Santiago Ayel', 'Fonseca Cantu Laura Isabel',
       'Pardo Leon Alison Tamara', 'Godoy Alonso Rafael',
       'Perez Hernandez Iker Daniel', 'Cruz Martinez Abelardo Gael',
       'Rocha Lopez Dara Sofia', 'Alvarez Garcia Mila',
       'Amezcua Larque Francisco Carlos', 'Espinoza Olivares Luis Samuel',
       'Perez Sanchez Leonardo David', 'Medina de la Cruz Axel Said',
       'Medina de la Cruz Hiram Karim', 'Lemos Fi

**Observacion :** En Mexico es comun usar dos apellidos, analizando y buscando por internet he llegado a esta conclusion, a diferencia que en Argentina, esto no suele ser comun, a la hora de aplicar este criterios en la columna de nombre de alumnos vamos a revisar si la nacionalidad de la personas es Mexico. Luego crearemos dos columnas para los nombres y apellidos de los alumnos.

Fuente: https://futbol.radioformula.com.mx/curiosidades/2022/8/13/por-que-en-mexico-usamos-dos-apellidos-61395.html

## Guardamos el archivo como csv y el nombre sera "Analisis y padres parcial"

In [211]:
df.to_csv('Alumnos y padres parcial.csv', index=False)